Userland/Libraries/LibWeb/HTML/Scripting/Environments.cpp

```diff
@@ -391,4 +393,33 @@ JS::Object& relevant_global_object(JS::Object const& object)
     return relevant_realm(object).global_object();
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#secure-context
+bool is_secure_context(Environment const& environment)
+{
+    // 1. If environment is an environment settings object, then:
+    if (is<EnvironmentSettingsObject>(environment)) {
+        // 1. Let global be environment's global object.
+        auto& global = static_cast<EnvironmentSettingsObject&>(const_cast<Environment&>(environment)).global_object();
```

Comment:
These casts feel weird. Mostly the const cast. Do we need a const global object getter?

Reply:
> Do we need a const global object getter?

Yep, the `Environment` virtual methods all seem to be non-const at the moment. Added a FIXME here and in one other place.

Reply:
(I'd rather do it this way than taking an `Environment&` and leaking it to the outside of this function though)